File: docs/legacy(old PACE)/C#/PaceAPISample/sample/CreateJob.cs
using System;
using System.Web.Services.Protocols;

using efipaceservices;

namespace Pace_Web_Service_SDK
{
	/// <summary>
	/// Summary description for CreateJob.
	/// </summary>
	public class CreateJob
	{

        private static string CustomerCode = "HOUSE";
        private static int QuantityOrdered = 2;

        public Job CreateJobForCustomer(string customerCode, int orderQty)
        {
            CustomerCode = customerCode;
            QuantityOrdered = orderQty;

            return Run();
        }

        public static Job Run()
		{
			CreateObjectPortType createObjectHttpBinding = PaceClient.getCreateObjectHttpBinding();

            if (CustomerExists(CustomerCode))
            {
                Job job = new Job();
                job.customer = CustomerCode;

                job = createObjectHttpBinding.createJob(job);

                // see how the default populated?
                Console.WriteLine("Created job " + job.job + " for '" + job.customer + "' on " + job.dateSetup.ToString());
                // update the Job PartQty Ordered
                UpdateJobPart(job, QuantityOrdered);
                return job;
            }
            else
            {
                return null;
            }
		}

		private static bool CustomerExists(string customerCode)
		{
            ReadObjectPortType readObjectHttpBinding = PaceClient.getReadObjectHttpBinding();			
			try
			{
				Customer cust	= new Customer();
				cust.id			= customerCode;
                readObjectHttpBinding.readCustomer(cust);
				return true;
			}
			catch (SoapException e)
			{
				Console.WriteLine("Customer: " + customerCode + " does not exist. Not adding job" + e.Message);
				return false;
			}
		}

		private static void UpdateJobPart(Job job, int quantity)
		{
            ReadObjectPortType readObjectHttpBinding = PaceClient.getReadObjectHttpBinding();
            UpdateObjectPortType updateObjectHttpBinding = PaceClient.getUpdateObjectHttpBinding();						

			JobPart part = new JobPart();
			part.job	 = job.job;
			part.jobPart = "01";
            part = readObjectHttpBinding.readJobPart(part);

			Console.WriteLine("Quantity Ordered " + quantity);
			part.qtyOrdered = quantity;
            updateObjectHttpBinding.updateJobPart(part);
			Console.WriteLine("Sent JobPart.qtyOrdered change request");

            part = readObjectHttpBinding.readJobPart(part);
			if ( quantity != part.qtyOrdered )
				throw new Exception("JobPart quantity update failed");
			Console.WriteLine("Read shows change JobPart.qtyOrdered: '" + part.qtyOrdered + "'");
		}
	}
}

File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/sample/CreateJob.vb
Imports PaceWebServiceSDK.efipaceservices
Imports System.Web.Services.Protocols

Public Class CreateJob

    Public Shared Function Run() As Job

        ' Inputs
        Console.WriteLine("Enter the customer code")
        Dim customerCode As String = Console.ReadLine()
        Dim quantityOrdered As Integer = 50

        ' 1. CreateObjectHttpBinding instance.
        Dim createObjectHttpBinding As CreateObjectHttpBinding = PaceClient.getCreateObjectHttpBinding()

        If CustomerExists(customerCode) Then
            ' 2. Create a new Instance of Job
            Dim newJob As Job = New Job()

            '3. Set the required fields
            newJob.customer = customerCode

            '4. Persist the data to EFI Pace System by calling createJob
            newJob = createObjectHttpBinding.createJob(newJob)
            Console.WriteLine("Created job " + newJob.job + " for '" + newJob.customer + "' on " + newJob.dateSetup)
            Return newJob
        Else
            Return Nothing
        End If

    End Function

    Private Shared Function CustomerExists(ByVal customerCode As String) As Boolean
        Dim customer As Customer = ReadCustomer.readCustomer(customerCode)
        Return customer IsNot Nothing
    End Function

End Class
